tests/integration/synapseclient/models/async/test_wiki_async.py

```diff
@@ -0,0 +1,751 @@
+"""Integration tests for the synapseclient.models.wiki module."""
+
+import gzip
+import os
+import tempfile
+import time
+import uuid
+from typing import Callable
+
+import pytest
+
+from synapseclient import Synapse
+from synapseclient.core.exceptions import SynapseHTTPError
+from synapseclient.models import (
+    Project,
+    WikiHeader,
+    WikiHistorySnapshot,
+    WikiOrderHint,
+    WikiPage,
+)
+
+
+@pytest.fixture(scope="session")
+async def wiki_page_fixture(
+    project_model: Project, syn: Synapse, schedule_for_cleanup: Callable[..., None]
+) -> WikiPage:
+    """Create a root wiki page fixture that can be shared across tests."""
+    wiki_title = f"Root Wiki Page"
+    wiki_markdown = "# Root Wiki Page\n\nThis is a root wiki page."
+
+    wiki_page = WikiPage(
+        owner_id=project_model.id,
+        title=wiki_title,
+        markdown=wiki_markdown,
+    )
+    root_wiki = await wiki_page.store_async(synapse_client=syn)
+    schedule_for_cleanup(root_wiki.id)
```

Comment:
Does this work? I know I had to add some specific logic for a few class types to work properly with the `clean` function